autode/calculations/executors.py

```diff
@@ -462,9 +463,11 @@ def _set_properties_from_optimiser(self) -> None:
         if final_coords is None:
             raise ex.CalculationException("Final coordinates undefined")
 
-        self.molecule.coordinates = final_coords.reshape((-1, 3))
-        if final_coords.g is not None:
-            self.molecule.gradient = final_coords.g.reshape((-1, 3))
+        cart_coords = final_coords.to("cart")
+        self.molecule.coordinates = cart_coords.reshape((-1, 3))
+        if cart_coords.g is not None:
+            self.molecule.gradient = cart_coords.g.reshape((-1, 3))
```

Comment:
I wonder if these reshapes should be added into the species coordinate/gradient setters. I bet there are other places where this is required. what do you think?

Reply:
@t-young31 Yes you are right. I checked the coordinate setter and it actually does a reshape. added the reshape into the gradient code.